File: Frontend/app/models/Event.js
import mongoose from "mongoose";
import { Schema, model, models } from "mongoose";

const eventSchema = new Schema({
  id: { type: String, required: true, unique: true },
  title: { type: String, required: true },
  description: { type: String, required: true },
  venue: { type: String, required: true },
  creator: { type: Schema.Types.ObjectId, ref: "User" },
  date: { type: Date, required: true },
  start_datetime: { type: String, required: true },
  end_datetime: { type: String, required: true },
  created_at: { type: Date, default: Date.now },
  required_skills: [{ type: Schema.Types.ObjectId, ref: "Skill" }],
  // Add other fields as necessary
});

const Event = models.Event || model("Event", eventSchema);

export default Event;

File: Backend/Zubin.Platform.WebApi/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Zubin.Platform.WebApi
{
    public class Event
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Venue { get; set; }
        public User Creator { get; set; }
        public DateTimeOffset Date { get; set; }
        public int quota { get; set; }
        public string StartDatetime { get; set; }
        public string EndDatetime { get; set; }
        public List<Skills> RequiredSkills { get; set; }
        public string CreatedAt { get; set; }
        public string UpdatedAt { get; set; }
    }
}
